File: src/main/java/lt/ltrp/vehicle/event/SpeedometerTickEvent.java
package lt.ltrp.vehicle.event;

import lt.ltrp.player.object.LtrpPlayer;
import lt.ltrp.vehicle.LtrpVehicle;
import net.gtaun.shoebill.event.player.PlayerEvent;

/**
 * @author Bebras
 *         2015.12.14.
 */
public class SpeedometerTickEvent extends PlayerEvent {

    private LtrpVehicle vehicle;
    private float speed;

    public SpeedometerTickEvent(LtrpPlayer player, LtrpVehicle vehicle, float speed) {
        super(player);
        this.vehicle = vehicle;
        this.speed = speed;
    }

    public float getSpeed() {
        return speed;
    }

    @Override
    public LtrpPlayer getPlayer() {
        return (LtrpPlayer)super.getPlayer();
    }

    public void setSpeed(float speed) {
        this.speed = speed;
    }

    public LtrpVehicle getVehicle() {
        return vehicle;
    }

    public void setVehicle(LtrpVehicle vehicle) {
        this.vehicle = vehicle;
    }
}

File: shoebill_plugins/vehicle-speedometer/src/main/kotlin/lt/ltrp/vehicle/speedometer/event/SpeedometerTickEvent.kt
package lt.ltrp.vehicle.speedometer.event

import lt.ltrp.vehicle.`object`.LtrpVehicle


/**
 * @author Bebras
 * *         2015.12.14.
 * Dispatched whenever a speedometer is updated
 * **This event is dispatched frequently**
 */
class SpeedometerTickEvent(vehicle: LtrpVehicle, val speed: Float) : SpeedometerEvent(vehicle) {

}
